File: Chap8-Internationalization/8-1InternationalizationDemo/main.cpp
#include "mainwindow.h"
#include <QApplication>
#include <QTranslator>
#include <QSettings>
#include <QDebug>

int main(int argc, char *argv[])
{
    QApplication a(argc, argv);

    QTranslator translator;
    QTranslator guiTranslator;


    QSettings settings("Blikoon Technologies", "Painter App");

    QString userLanguage = settings.value("language","english").toString();

    QString systemLang = QLocale::system().name();

    qDebug() << "Sytem language is :" << systemLang;

    if(userLanguage == "default"){

        if(systemLang == "zh_CN"){
            translator.load(":/translations/chinese.qm");
            guiTranslator.load(":/translations/qt_zh_CN_official.qm");
            a.installTranslator(&translator);
            a.installTranslator(&guiTranslator);
        }

        if(systemLang == "fr_FR"){
            translator.load(":/translations/french.qm");
            guiTranslator.load(":/translations/qt_fr_FR_official.qm");
            a.installTranslator(&translator);
            a.installTranslator(&guiTranslator);

        }

    }

    if(userLanguage == "english"){
        //English loaded by default
    }

    if(userLanguage == "french"){
        translator.load(":/translations/french.qm");
        guiTranslator.load(":/translations/qt_fr_FR_official.qm");

    }

    if(userLanguage == "chinese"){
        translator.load(":/translations/chinese.qm");
        guiTranslator.load(":/translations/qt_zh_CN_official.qm");
    }

    if((userLanguage == "french") || (userLanguage == "chinese")){
           a.installTranslator(&translator);
           a.installTranslator(&guiTranslator);
       }

    MainWindow w;
    w.show();

    return a.exec();
}

File: Chap8-Internationalization/8-1InternationalizationDemo/main.py
import sys
from PySide6.QtWidgets import QApplication
from PySide6.QtCore import QTranslator, QSettings, QLocale
from PySide6.QtGui import QScreen
from mainwindow import MainWindow

def main():
    app = QApplication(sys.argv)
    
    # Set up translators
    translator = QTranslator()
    gui_translator = QTranslator()
    
    # Get language settings
    settings = QSettings("Blikoon Technologies", "Painter App")
    user_language = settings.value("language", "english")
    
    system_lang = QLocale.system().name()
    print(f"System language is: {system_lang}")
    
    # Load appropriate translation based on settings
    if user_language == "default":
        if system_lang == "zh_CN":
            translator.load(":/translations/chinese.qm")
            gui_translator.load(":/translations/qt_zh_CN_official.qm")
            app.installTranslator(translator)
            app.installTranslator(gui_translator)
        
        if system_lang == "fr_FR":
            translator.load(":/translations/french.qm")
            gui_translator.load(":/translations/qt_fr_FR_official.qm")
            app.installTranslator(translator)
            app.installTranslator(gui_translator)
    
    elif user_language == "english":
        # English loaded by default
        pass
    
    elif user_language == "french":
        translator.load(":/translations/french.qm")
        gui_translator.load(":/translations/qt_fr_FR_official.qm")
        app.installTranslator(translator)
        app.installTranslator(gui_translator)
    
    elif user_language == "chinese":
        translator.load(":/translations/chinese.qm")
        gui_translator.load(":/translations/qt_zh_CN_official.qm")
        app.installTranslator(translator)
        app.installTranslator(gui_translator)
    
    window = MainWindow()
    window.show()
    
    return app.exec()

if __name__ == "__main__":
    sys.exit(main())